plugins/command-manager/src/main/java/com/wazuh/commandmanager/rest/RestPostCommandAction.java

```diff
@@ -180,4 +160,75 @@ private RestChannelConsumer handlePost(RestRequest request) throws IOException {
                             });
         };
     }
+
+    /**
+     * Parses the content of a RestRequest and retrieves a list of Command objects.
+     *
+     * @param request the RestRequest containing the command data.
+     * @return a list of Command objects parsed from the request content.
+     * @throws IOException if an error occurs while parsing the request content.
+     */
+    private static List<Command> getCommandList(RestRequest request) throws IOException {
+        // Request parsing
+        XContentParser parser = request.contentParser();
+        List<Command> commands = new ArrayList<>();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+        parser.nextToken();
+        if (parser.nextToken() == XContentParser.Token.START_ARRAY) {
+            commands = Command.parseToArray(parser);
+        } else {
+            log.error("Token does not match {}", parser.currentToken());
+        }
+
+        return commands;
+    }
```

Comment:
Parsing should be done in the model classes only.

Reply:
Fixed on commit de10364e17b61e068b2c0ac7be853ff2f8328436